src/h5web/visualizer/utils.ts

```diff
@@ -1,6 +1,18 @@
 import type { HDF5Entity, HDF5Metadata } from '../providers/models';
 import { VIS_DEFS, Vis } from '../visualizations';
 
+function applySpecialVisRules(visArr: Vis[]): Vis[] {
+  if (
+    // Remove `Raw` or `NxSpectrum` vis if any other vis is supported
+    (visArr[0] === Vis.Raw || visArr[0] === Vis.NxSpectrum) &&
+    visArr.length > 1
+  ) {
+    return visArr.slice(1);
+  }
+
+  return visArr;
+}
+
```

Comment:
This is where we remove `NxSpectrum` from the array of supported visualizations when `NxImage` is also supported.

Reply:
How can we be sure that `Raw` or `NxSpectrum` will always be in first position ?

Reply:
Because the vis definitions are ordered. I considered using a more complex algorithm, but thought it might over-engineering. We have tests to check that `Raw` and `NxSpectrum` are indeed removed, so if we re-order the vis definitions, it will catch the regression.

Reply:
Not a fan of relying on this ordering but as long as we can catch the regression...

Reply:
Yeah... Though the ordering is critical visually but also for the default vis selection, so this function is not the only thing that's coupled to it.